internal/epoll/poller.go

```diff
@@ -154,13 +176,22 @@ func (p *Poller) Wait(events []unix.EpollEvent, deadline time.Time) (int, error)
 			return 0, fmt.Errorf("epoll wait: %w", os.ErrDeadlineExceeded)
 		}
 
-		for _, event := range events[:n] {
-			if int(event.Fd) == p.event.raw {
-				// Since we don't read p.event the event is never cleared and
+		for i := 0; i < n; {
+			event := events[i]
+			if int(event.Fd) == p.closeEvent.raw {
+				// Since we don't read p.closeEvent the event is never cleared and
 				// we'll keep getting this wakeup until Close() acquires the
 				// lock and sets p.epollFd = -1.
 				return 0, fmt.Errorf("epoll wait: %w", os.ErrClosed)
 			}
+			if int(event.Fd) == p.flushEvent.raw {
+				// read event to prevent it from continuing to wake
+				p.flushEvent.read()
+				events = slices.Delete(events, i, i+1)
+				n -= 1
+				continue
```

Comment:
So the behaviour is that `Wait` will return `0, nil` for a flush. But what about a case where another ring becomes ready simultaneously with flushEvent? In that case we'd get two events from epoll, would discard one and then `return 1, nil`. In that case we drop an `ErrFlushed` from `Reader.Read` which the user is expecting.

How about removing the event here, and doing `return n, ErrFlushed` from `Wait` instead? Then `ringbuf` and `perf` re-export `epoll.ErrFlushed` and the rest can mostly stay the same.

Reply:
> In that case we drop an ErrFlushed from Reader.Read which the user is expecting.
I don't follow. `Flush` is the one that sets up a pending error of `ErrFlushed`, which will still be returned after all the rings have been read.

The `n` return value from `Wait` isn't even used. Only if an error is returned does a different code path happen. We want `Flush` to behave as if a regular ring epoll event happened, which is what it is doing.

Reply:
I think we actually want to ensure we read all the events in case there both a `Flush` and `Close` in the same wakeup.